DataConnectors/Trend Micro/AzureFunctionTrendMicroXDR/TimerTrigger/__init__.py

```diff
@@ -0,0 +1,143 @@
+            hostname = impact_scope['entityValue']['name']
+            host_guid = impact_scope['entityValue']['guid']
+        else: scope_value = impact_scope['entityValue']
+
+
+        simple_str_field = 'impactScope_' + entity_type
+        if simple_str_field not in json2_data:  # use first element only
+            json2_data[simple_str_field] = scope_value
+        if entity_type == 'host':
+            json2_data['impactScope_hostname'] = hostname
+            json2_data['impactScope_hostGuid'] = host_guid
+
+        comma_seperated_field = 'impactScope_' + entity_type + 's'
+        if comma_seperated_field not in json2_data:
+            json2_data[comma_seperated_field] = scope_value
+        else:
+            json2_data[comma_seperated_field] = "{}, {}".format(json2_data[comma_seperated_field], scope_value)
+
+    return json2_data
+
+# Azure Provided Code for posting data to Azure Log Ingestion
+def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
+    x_headers = 'x-ms-date:' + date
+    string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
+    bytes_to_hash = bytes(string_to_hash, encoding='utf-8') 
+    decoded_key = base64.b64decode(shared_key)
+    encoded_hash = base64.b64encode(hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()).decode()
+    authorization = "SharedKey {}:{}".format(customer_id,encoded_hash)
+    return authorization
+
+# Required Function to create and invoke an API POST request to the Azure Log Analytics Data Collector API. Reference: https://docs.microsoft.com/azure/azure-functions/functions-reference-python#environment-variables
+
+def post_data(customer_id, shared_key, body, log_type, workbencheIds):
+    method = 'POST'
+    content_type = 'application/json'
+    resource = '/api/logs'
+    rfc1123date = datetime.datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')
+    content_length = len(body)
+    signature = build_signature(customer_id, shared_key, rfc1123date, content_length, method, content_type, resource)
+    uri = 'https://' + customer_id + '.ods.opinsights.azure.com' + resource + '?api-version=2016-04-01'
+
+    headers = {
+        'content-type': content_type,
+        'Authorization': signature,
+        'Log-Type': log_type,
+        'x-ms-date': rfc1123date
+    }
+
+    response = requests.post(uri,data=body, headers=headers)
+    if (response.status_code >= 200 and response.status_code <= 299):
+        print ('Accepted ' + workbencheIds)
+    #Uncomment for easy troublshooting of log posting to Sentinel 
+        #print (body)
+    else:
+        print (response)
+
+
+# Main Code: If inital list events contains new workbenchs, loop through all new workbenchs and pull data and forward to Sentinal API
+def function():
+    workbench_list = getWorkbenchList()
+    a = 0
+    if "totalCount" in workbench_list['data']:

```

Comment:
Can you please take care of the scenario when getWorkbenchList() returns empty workbench_list. 
We see **Result: Failure Exception: KeyError: 'data'** when api returns empty.

Reply:
Actually that is due to API auth failure..  Will address.